src/libparms/db/parmsdb.h

```diff
@@ -122,24 +122,18 @@ class PARMS_EXPORT ParmsDb : public Db {
         bool insertMigrationSelectiveSync(const MigrationSelectiveSync &migrationSelectiveSync);
         bool selectAllMigrationSelectiveSync(std::vector<MigrationSelectiveSync> &migrationSelectiveSyncList);
 
-        bool selectLastServerSelfRestartTime(int64_t &lastServerRestartTime);
-        bool selectLastClientSelfRestartTime(int64_t &lastClientRestartTime);
-        bool updateLastServerSelfRestartTime(int64_t lastServerRestartTime = -1 /* -1 means now*/);
-        bool updateLastClientSelfRestartTime(int64_t lastClientRestartTime = -1 /* -1 means now*/);
-
         bool selectAppState(AppStateKey key, std::string &value, bool& found);
         bool updateAppState(AppStateKey key, const std::string &value, bool& found);  // update or insert
 
-       
     private:
         static std::shared_ptr<ParmsDb> _instance;
         bool _test;
 
         ParmsDb(const std::filesystem::path &dbPath, const std::string &version, bool autoDelete, bool test);
 
         bool insertDefaultParameters();
-        bool insertDefaultSelfRestarterData();
         bool insertDefaultAppState();
+        bool insertDefaultAppState(AppStateKey key, const std::string &value);
```

Comment:
not very clear to me why this is declared in `parmsdb.h` but defined in `parmsdbappstate.cpp` while there is a `parmsdbappstate.h`.
Is it possible to make ParmsDbAppState as a separated class, perhaps a child of ParmsDb is there is some methods defined in ParmsDb that you need in ParmsDbAppState

Reply:
As discussed with @luc-guyot-infomaniak in previous PR ([link to discussion](https://github.com/Infomaniak/desktop-kDrive/pull/76#discussion_r1580962129)), the goal here is to reduce the size of `parmsdb.cpp`, which is already more than 3000 lines long.

This reduction will improve the readability of `parmsdb` functions. For instance:

```cpp
ASSERT(queryCreate(CREATE_APP_STATE_TABLE_ID));
if (!queryPrepare(CREATE_APP_STATE_TABLE_ID, CREATE_APP_STATE_TABLE, false, errId, error))
{
    queryFree(CREATE_APP_STATE_TABLE_ID);
    return sqlFail(CREATE_APP_STATE_TABLE_ID, error);
}
if (!queryExec(CREATE_APP_STATE_TABLE_ID, errId, error)) {
    queryFree(CREATE_APP_STATE_TABLE_ID);
    return sqlFail(CREATE_APP_STATE_TABLE_ID, error);
}
queryFree(CREATE_APP_STATE_TABLE_ID);

//repeat for all tables
```

will become:

```cpp
createAppState();
createOtherTable();
...
```

The same simplification will apply to the prepare function.

Currently, only `parmsdbappstate.h/cpp` are done, but the plan is to do the same with other parts of `parmsdb.cpp` (e.g., `insertDrive()`, `updateDrive()`, `deleteDrive()` -> `parmsdbdrive.h/cpp`).

Making it a separate or child class seems to be quite complicated as we are still within the same database, which is protected with a mutex (this would require making the mutex a protected static member). Additionally, there's the question of database creation/upgrade, which needs to be synchronized for all potential child classes.

However, what could be done is to remove `parmsdbappstate.h`, whose only goal is to define constants (all the requests), but this can also be done in the cpp file, it was just to separate the sql queries definition.